homeassistant/components/websocket_api/commands.py

```diff
@@ -196,3 +198,37 @@ def handle_ping(hass, connection, msg):
     Async friendly.
     """
     connection.send_message(pong_message(msg['id']))
+
+
+@callback
+@decorators.websocket_command({
+    vol.Required('type'): 'render_template',
+    vol.Required('template'): cv.template,
+    vol.Optional('entity_ids'): cv.entity_ids
+})
+def handle_render_template(hass, connection, msg):
+    """Handle render_template command.
+
+    Async friendly.
+    """
+    template = msg['template']
+    if template is not None:
+        template.hass = hass
+
+        entity_ids = msg.get('entity_ids')
+        if entity_ids is None:
+            entity_ids = list(set(template.extract_entities()))
+
+        @callback
+        def state_listener(entity, old_state, new_state):
+            connection.send_message(messages.event_message(msg['id'], {
+                'result': template.async_render(),
+                }))
+
+        connection.subscriptions[msg['id']] =
+            async_track_state_change(hass, entity_ids, state_listener)
```

Comment:
You should probably want to use `async_track_template` from `helpers/event.py`

Reply:
That doesn't allow for manual entity tracking (though I could add it while I'm at it).

I foresee people using this for things like tracking the number of sensors that are triggered and such - like looping over all entities of a given domain. That would not work with the automatic entity extraction.
The fact that it would be better to use a template sensor for things like that doesn't mean people will, unfortunately...

Reply:
actually, our new template tracking will work with looping over domains. See #23590

Reply:
I guess we can change the implementation once this is merged though

Reply:
Looking at it now, that's not at all what `async_track_template` does.

From what I can tell, it runs an action *once* once the template changes to evaluate `true`...